### DISCUSSION TITLE
Offering assistance for "JS app event hooks with goja and update the jsvm bindings"

### DISCUSSION THREAD

**Discussion by MarNeumann** [pocketbase/pocketbase#2504] [Q&A] [answered]
Hi everyone,

I've been following the development roadmap of PocketBase closely and noticed the activity "JS app event hooks with goja and update the jsvm bindings" right at the top of the backlog. As someone who is eager to see this feature come to life, I'm highly interested in contributing to its implementation. I assume that this particular activity involves a significant amount of repetitive yet necessary steps.

I would like to offer my assistance in tackling these repetitive tasks and making progress on the implementation. My idea is that once one hook is successfully implemented, I can replicate the process for similar hooks, thereby accelerating the overall development.

Could anyone provide some guidance on how I can best contribute to the project in this regard? I'm particularly interested in understanding the specific steps involved in implementing the JS app event hooks with goja and updating the jsvm bindings. Are there any existing examples or documentation that I can refer to in order to familiarize myself with the expected workflow and requirements?

Thank you for considering my offer to contribute, and I'm looking forward to your guidance and feedback.

Best regards,
codama

**Comment by ganigeorgiev** [comment, marked as answer]
Thank you but at least for now I don't need help with the tasks from the roadmap since I still don't have clear definition how things should work and I want to experiment with different approaches before choosing the final api interface.

Before introducing the JS app hooks we'll also need first to refactor and simplify some of the internals as it starting to become too overwhelming and hard to keep track of everything in your head (_there will be an announcement somewhere after the upcoming v0.16.0 release with more info about the planned changes_).
The difficult and most time consuming part with the JS app hooks is the documentation, not the actual code as the heavy lifting will be done by [goja](https://github.com/dop251/goja) and for most of the things I expect to work right out of the box, similar to how currently the JS migrations works.

**Comment by MarNeumann** [comment]
That sounds great! When the time comes for documentation work on the JS app hooks, I'm more than happy to contribute in that area as well. Just let me know how I can be of assistance, and I'll gladly lend a hand in creating clear and comprehensive documentation for the feature.

Looking forward to the upcoming changes!
